src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -591,9 +591,7 @@ public static PSDataCollection<PSObject> GetPSDataCollection(object inputValue)
         /// <param name="second">Object to compare first to.</param>
```

Comment:
@vexx32, your last commit had 2 failures in `PowerShell-CI-windows`
SemanticVersion api tests.Error handling.Missing parts: 'NullString': '$null'
```powershell
Expected an exception, with FullyQualifiedErrorId 'FormatException' to be thrown, but the FullyQualifiedErrorId was 'PSArgumentNullException'. from D:\a\1\s\test\powershell\engine\Basic\SemanticVersion.Tests.ps1:209 char:19
    +                 { [SemanticVersion]::Parse($version) } | Should -Thro ...
    +                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\SemanticVersion.Tests.ps1: line 209
209:                 { [SemanticVersion]::Parse($version) } | Should -Throw -ErrorId "FormatException"
```
Null Representatives.Casting Behaviour.Error occurred in Context block
```powershell
Unable to find type [System.Management.Automation.AutomationNull].
At D:\a\1\s\test\powershell\Language\Scripting\NullRepresentatives.Tests.ps1:51 char:28
```

Reply:
@daxian-dbw looking at the tests... It almost looks like this is working as intended?

Am I missing something there? Seems to me as though `ArgumentNullException` would be pretty much exactly what I'd expect a `NullString.Value` input to yield. 😕

Reply:
Ah, found the issue. The test's check for $null assumes that NullString.Value wouldn't pass the test -- but not it does. Fixed!